### PR TITLE
crane-gps-watch: init at 20160308

### PR DESCRIPTION
**Pull request by AcouBass**
###### Motivation for this change

New pkg -  small piece of software for creating .gpx files from various GPS watches. 
###### Things done
- [X] Tested using sandboxing
  ([nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS,
    or option `build-use-chroot` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [X] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/misc/crane-gps-watch/default.nix [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  name = "crane-gps-watch";
+  version = "20160308";
+  src = fetchFromGitHub {
+    owner = "mru00";
+    repo = "crane_gps_watch";
+    rev = "221f99dc2925a07658f7b6e01789ced1a18cc08c";
+    sha256 = "0ji50yg85a114pdvw8k5jgfxq312frlxq67bazfaws9mkhnkpi9v";
+  };
+
+meta = {
```
Comment (joachifm) [inline pkgs/applications/misc/crane-gps-watch/default.nix]: Please consider adding yourself as the maintainer.

**Comment by AcouBass** [comment]
Added myself as maintainer in the package and in lib/maintainers.nix.

File: pkgs/applications/misc/crane-gps-watch/default.nix [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  version = "20160308";
+  src = fetchFromGitHub {
+    owner = "mru00";
+    repo = "crane_gps_watch";
+    rev = "221f99dc2925a07658f7b6e01789ced1a18cc08c";
+    sha256 = "0ji50yg85a114pdvw8k5jgfxq312frlxq67bazfaws9mkhnkpi9v";
+  };
+
+meta = with stdenv.lib; {
+  homepage = "https://www.github.com/mru00/crane_gps_watch";
```
Comment (joachifm) [inline pkgs/applications/misc/crane-gps-watch/default.nix]: Equivalently, `inherit (src.meta) homepage`

File: pkgs/applications/misc/crane-gps-watch/default.nix [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  description = "Crane GPS Watch client for Linux and Windows";
```
Comment (joachifm) [inline pkgs/applications/misc/crane-gps-watch/default.nix]: I think mentioning platforms in the description is redundant, given `meta.platforms` --- and in this case it is also inaccurate. It doesn't help the user to know that the software works on Windows, if the expression will only evaluate on Linux.

File: pkgs/applications/misc/crane-gps-watch/default.nix [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{ stdenv, fetchFromGitHub, libxml2 }:
+
+stdenv.mkDerivation rec {
+  name = "crane-gps-watch";
```
Comment (joachifm) [inline pkgs/applications/misc/crane-gps-watch/default.nix]: Please make `version` part of the name.

**Comment by joachifm** [comment]
Added a few more notes. When they have been addressed, please rebase and squash the commits into one.

File: pkgs/applications/misc/crane-gps-watch/default.nix [hunk 5]
```diff
@@ -0,0 +1,20 @@
+{ stdenv, fetchFromGitHub, libxml2 }:
+
+stdenv.mkDerivation rec {
+  name = "crane-gps-watch-${version}";
+  version = "221f99dc2925a07658f7b6e01789ced1a18cc08c";
```
Comment (joachifm) [inline pkgs/applications/misc/crane-gps-watch/default.nix]: Versions ought to satisfy `new > old`, so ISO 8601 dates are a better choice for vcs checkouts.

File: pkgs/applications/misc/crane-gps-watch/default.nix [hunk 6]
```diff
@@ -0,0 +1,20 @@
+{ stdenv, fetchFromGitHub, libxml2 }:
+
+stdenv.mkDerivation rec {
+  name = "crane-gps-watch-${version}";
+  version = "20160308";
+  src = fetchFromGitHub {
+    owner = "mru00";
+    repo = "crane_gps_watch";
+    rev = "$221f99dc2925a07658f7b6e01789ced1a18cc08c";
```
Comment (joachifm) [inline pkgs/applications/misc/crane-gps-watch/default.nix]: The `$` at the beginning of the `rev` looks like a typo.

File: pkgs/applications/misc/crane-gps-watch/default.nix [hunk 7]
```diff
@@ -0,0 +1,20 @@
+{ stdenv, fetchFromGitHub, libxml2 }:
```
Comment (joachifm) [inline pkgs/applications/misc/crane-gps-watch/default.nix]: `libxml2` is unused.

**Comment by joachifm** [comment]
Please squash into a single commit.

**Comment by Mic92** [comment]
I changed the version number. There is a python2 script, which requires vte with python support and pygtk.

```patch
From e4b07527ab892ce5d362cc2798bdab941c57f2a9 Mon Sep 17 00:00:00 2001
From: Edd Baxter <pieceredd@gmail.com>
Date: Sun, 21 Aug 2016 01:39:03 +0100
Subject: [PATCH] crane-gps-watch: init at 2016-03-08
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Jörg Thalheim <joerg@higgsboson.tk>
---
 lib/maintainers.nix                                |  1 +
 pkgs/applications/misc/crane-gps-watch/default.nix | 20 ++++++++++++++++++++
 pkgs/top-level/all-packages.nix                    |  2 ++
 3 files changed, 23 insertions(+)
 create mode 100644 pkgs/applications/misc/crane-gps-watch/default.nix

diff --git a/lib/maintainers.nix b/lib/maintainers.nix
index 08d6c8a..a2790c2 100644
--- a/lib/maintainers.nix
+++ b/lib/maintainers.nix
@@ -12,6 +12,7 @@
   abbradar = "Nikolay Amiantov <ab@fmap.me>";
   aboseley = "Adam Boseley <adam.boseley@gmail.com>";
   abuibrahim = "Ruslan Babayev <ruslan@babayev.com>";
+  AcouBass = "Edd Baxter <pieceredd@gmail.com>";
   acowley = "Anthony Cowley <acowley@gmail.com>";
   adev = "Adrien Devresse <adev@adev.name>";
   Adjective-Object = "Maxwell Huang-Hobbs <mhuan13@gmail.com>";
diff --git a/pkgs/applications/misc/crane-gps-watch/default.nix b/pkgs/applications/misc/crane-gps-watch/default.nix
new file mode 100644
index 0000000..d637478
--- /dev/null
+++ b/pkgs/applications/misc/crane-gps-watch/default.nix
@@ -0,0 +1,20 @@
+{ stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  name = "crane-gps-watch-${version}";
+  version = "2016-03-08";
+  src = fetchFromGitHub {
+    owner = "mru00";
+    repo = "crane_gps_watch";
+    rev = "221f99dc2925a07658f7b6e01789ced1a18cc08c";
+    sha256 = "0ji50yg85a114pdvw8k5jgfxq312frlxq67bazfaws9mkhnkpi9v";
+  };
+
+  meta = with stdenv.lib; {
+    inherit (src.meta) homepage;
+    description = "Crane GPS Watch client";
+    license = stdenv.lib.licenses.gpl3;
+    maintainers = [ maintainers.AcouBass ];
+    platforms = platforms.linux;
+  };
+}
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index 5132a67..92ffc3c 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -748,6 +748,8 @@ in
 
   cpulimit = callPackage ../tools/misc/cpulimit { };
 
+  crane-gps-watch = callPackage ../applications/misc/crane-gps-watch { };
+
   contacts = callPackage ../tools/misc/contacts { };
 
   coturn = callPackage ../servers/coturn { };
-- 
2.10.0

```

**Comment by joachifm** [comment]
@AcouBass ping?

**Comment by joachifm** [comment]
Closing as abandoned.

**Pull request closed without merging**